### PR TITLE
[INLONG-11037][SDK] Transform support ENCODE() and DECODE() function

### LINKED ISSUE #11037: [Feature][SDK] Transform support ENCODE() and DECODE() function
### Description

* parent issue: #10796 
* from flink sql: https://nightlies.apache.org/flink/flink-docs-release-1.17/zh/docs/dev/table/functions/systemfunctions/#%E7%AE%97%E6%9C%AF%E5%87%BD%E6%95%B0
* DECODE(binary, string)--Decode using the supplied character set (' US-ASCII ', 'ISO-8859-1', 'UTF-8', 'UTF-16BE', 'UTF-16LE', 'UTF-16'). If either parameter is empty, the result will also be empty.
* ENCODE(string1, string2)--Encode using the provided character set (' US-ASCII ', 'ISO-8859-1', 'UTF-8', 'UTF-16BE', 'UTF-16LE', 'UTF-16'). If either parameter is empty, the result will also be empty.

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by emptyOVO**
Fixes #11037 

### Motivation

add DecodeFunction class and EncodeFunction class, provide unit tests
### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [x] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/DecodeFunction.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        Object characterObj = characterSetParser.parse(sourceData, rowIndex, context);
+        if (binaryObj == null || characterObj == null) {
+            return null;
+        }
+        String binaryString = OperatorTools.parseString(binaryObj);
+        String characterSetValue = OperatorTools.parseString(characterObj).toUpperCase();
+        return decode(binaryString, characterSetValue);
+    }
+
+    private String decode(String binaryString, String charsetName) {
```
Comment (luchunliang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/DecodeFunction.java:57]: ﻿Encode&decode is a broad concept that requires a clear definition of the specific encoding type

Comment (emptyOVO) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/DecodeFunction.java:75]: done

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/DecodeFunction.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+import net.sf.jsqlparser.expression.Function;
+
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@TransformFunction(names = {"decode"})
```
Comment (vernedeng) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/DecodeFunction.java:35]: please add the documentation of these two functions

Comment (emptyOVO) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/DecodeFunction.java:40]: done

**Review by aloyszhang — APPROVED** [review]

**Review by luchunliang — APPROVED** [review]

**Pull request merged**